docs/core/tools/dotnet-sln.md

```diff
@@ -32,61 +31,141 @@ To use the `dotnet sln` command, the solution file must already exist. If you ne
 dotnet new sln
 ```
 
+## Arguments
+
+- **`SOLUTION_FILE`**
+
+  The solution file to use. If not specified, the command searches the current directory for one. If there are multiple solution files in the directory, one must be specified.
+
+## Options
+
+- **`-h|--help`**
+
+  Prints out a short help for the command.
+
 ## Commands
 
-`add <PROJECT> ...`
+### `add`
+
+Adds a project or multiple projects to the solution file.
+
+#### Synopsis
+
+```dotnetcli
+dotnet sln [<SOLUTION_FILE>] add [--in-root] [-s|--solution-folder] <PROJECT_PATH>
+dotnet sln add [-h|--help]
+```
+
+#### Arguments
+
+- **`SOLUTION_FILE`**
+
+  The solution file to use. If not specified, the command searches the current directory for one. If there are multiple solution files in the directory, one must be specified.
+
+- **`PROJECT_PATH`**
+
+  The path to the project to add to the solution. Add multiple projects by adding one after the other separated by spaces. [Globbing patterns](https://en.wikipedia.org/wiki/Glob_(programming)) are also supported on Unix/Linux-based terminals.
+
+#### Options
+
+- **`-h|--help`**
+
+  Prints out a short help for the command.
+
+- **`--in-root`**
+
+  Places the projects in the root of the solution, rather than creating a solution folder. Available since .NET Core 3.0 SDK.
+
+- **`-s|--solution-folder`**
+
+  The destination solution folder path to add the projects to. Available since .NET Core 3.0 SDK.
 
-`add <GLOBBING_PATTERN>`
+### `remove`
 
-Adds a project or multiple projects to the solution file. [Globbing patterns](https://en.wikipedia.org/wiki/Glob_(programming)) are supported on Unix/Linux based terminals.
+Removes a project or multiple projects from the solution file.
 
-`remove <PROJECT> ...`
+#### Synopsis
 
-`remove <GLOBBING_PATTERN>`
+```dotnetcli
+dotnet sln [<SOLUTION_FILE>] remove <PROJECT_PATH>
+dotnet sln [<SOLUTION_FILE>] remove [-h|--help]
+```
+
+#### Arguments
+
+- **`SOLUTION_FILE`**
+
+  The solution file to use. If not specified, the command searches the current directory for one. If there are multiple solution files in the directory, one must be specified.
+
+- **`PROJECT_PATH`**
+
+  The path to the project to remove from the solution. Remove multiple projects by adding one after the other separated by spaces. [Globbing patterns](https://en.wikipedia.org/wiki/Glob_(programming)) are also supported on Unix/Linux-based terminals.
```

Comment:
I'm concerned about the sentence on globbing. My understanding is that the unix shell expands glob patterns before invoking the `dotnet` command. Adding this statement makes me wonder if globbing is implemented in the command. If it was, I would think I should quote the pattern:  `dotnet sln remove '*.*'`. I don't that's what you meant. If globbing is performed by the shell, I'd reword it as follows:

```suggestion
  The path to the project to remove from the solution. Remove multiple projects by adding one after the other separated by spaces. Unix/Linux shell [globbing pattern](https://en.wikipedia.org/wiki/Glob_(programming)) expansions are processed correctly by the `dotnet sln` command.
```

Reply:
Yes, they're performed by the shell, not by the CLI.